src/app/templates/AssetInfo.tsx

```diff
@@ -32,14 +30,14 @@ const AssetInfo: FC<AssetInfoProps> = ({ assetSlug }) => {
         id="contract-address"
         label={<T id="contract" />}
         labelDescription={<T id="addressOfTokenContract" substitutions={[getAssetSymbol(metadata)]} />}
-        value={asset.contract}
+        value={isTezAsset(asset) ? 'TEZ' : asset.contract}
         size={36}
         style={{
           resize: 'none'
         }}
       />
 
-      {isFA2Token(asset) && (
+      {!isTezAsset(asset) && isFA2Token(asset) && (
```

Comment:
this's weird that `isFA2Token` fn returns `true` for `XTZ` asset.  
move this check inside `isFA2Token`